backend/test_observer/data_access/models_enums.py

```diff
@@ -25,6 +25,7 @@
 class FamilyName(str, Enum):
     SNAP = "snap"
     DEB = "deb"
+    CHARM = "charm"
```

Comment:
No modifications to `models.py`. The `snap_unique` constraint applies to Charms as well (unique on name, version, and track).

However, I would prefer if there was a single unique constraint that applied to all artefacts, would a unique contraint on name, version, and family be adequate?

It's tough to do with the current implementation, but it is possible to add `family_id` as an attribute to `Artefact`, and then a foreign key in `Artefact` on `(Stage.id, Stage.family_id`), and then a unique constraint on `Artefact` for `(name, version, family_id)`.

The caveat there is that I don't believe `Artefact.family_id` is auto populated unless we add an SQL trigger or there's a way that sqlalchemy can, but I didn't look into it that far.